src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -253,25 +253,25 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
     public constructor(props: ISendMessageComposerProps, context: React.ContextType<typeof RoomContext>) {
         super(props, context);
 
-        if (this.props.mxClient.getCrypto() && this.props.mxClient.isRoomEncrypted(this.props.room.roomId)) {
-            this.prepareToEncrypt = throttle(
-                () => {
-                    this.props.mxClient.getCrypto()?.prepareToEncrypt(this.props.room);
-                },
-                60000,
-                { leading: true, trailing: false },
-            );
-        }
-
         const partCreator = new CommandPartCreator(this.props.room, this.props.mxClient);
         const parts = this.restoreStoredEditorState(partCreator) || [];
         this.model = new EditorModel(parts, partCreator);
         this.sendHistoryManager = new SendHistoryManager(this.props.room.roomId, "mx_cider_history_");
     }
 
-    public componentDidMount(): void {
+    public async componentDidMount(): Promise<void> {
         window.addEventListener("beforeunload", this.saveStoredEditorState);
         this.dispatcherRef = dis.register(this.onAction);
+
+        if (await this.props.mxClient.getCrypto()?.isEncryptionEnabledInRoom(this.props.room.roomId)) {
+            this.prepareToEncrypt = throttle(
+                () => {
+                    this.props.mxClient.getCrypto()?.prepareToEncrypt(this.props.room);
+                },
+                60000,
+                { leading: true, trailing: false },
+            );
+        }
```

Comment:
What happens if `isEncryptionEnabledInRoom` is still resolving by the time the user types something, encryption won't be prepared it looks like, this is undesirable and a regression